### PR TITLE
Add support for key value yarn.tags

### PR DESCRIPTION
**Pull request by mchataigner**
It may be interesting to tag via key value applications running on a hadoop cluster.
Unfortunately, yarn.tags (supported by most frameworks) are a list of string.
This change adds the ability to have special yarn.tags containing a ':' to split key and value.

All those tags will appear in the ApplicationEvent under field user_tags which is a map of
string to string.

It will be exported as nested object on ElasticSearch or Parquet.

### CODE REVIEW

**Comment by mchataigner** [comment]
> please also update readers/elasticsearch/src/main/resources/template.json to add user_tags field description:
> ` "user_tags": { "type": "object" },`
> 
> I'm still not convinced on the real need of this new field. yarn_tags seems enough to me.
> If we move forward with this field is there any way to ensure that number of key won't leads to too many fields being created in ES.

Done

File: agent/src/main/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/resourcemanager/RMContextImplEventRunnable.java [hunk 1]
```diff
@@ -49,23 +61,23 @@ public void sendAppEvent(ApplicationId applicationId, RMApp rmApp) {
...
-                    .filter(tag -> !tag.contains(":"))
+                    .filter(tag -> YARN_TAGS_TO_EXTRACT.stream().noneMatch(tag::startsWith))
                     .forEach(eventBuilder::addYarnTags);
 
             rmApp.getApplicationTags().stream()
-                    .filter(tag -> tag.contains(":"))
+                    .filter(tag -> YARN_TAGS_TO_EXTRACT.stream().anyMatch(tag::startsWith))
                     .map(tag -> {
                         int idx = tag.indexOf(':');
                         String key = tag.substring(0, idx);
```
Comment (ashangit) [inline agent/src/main/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/resourcemanager/RMContextImplEventRunnable.java]: will failed if ":" is not found on tag string as idx will be -1

File: readers/elasticsearch/src/main/resources/template.json [hunk 2]
```diff
@@ -65,6 +65,18 @@
         }
       }
     },
+    "project_name": {
```
Comment (ashangit) [inline readers/elasticsearch/src/main/resources/template.json]: please apply same config than the one for application_name field. The idea here is to have a project_name.keyword and workflow_name.keyword fields which are fields without split and normalisation 
"application_name": {
      "type": "text",
      "norms": false,
      "index_options": "positions",
      "fields": {
        "keyword": {
          "type": "keyword",
          "ignore_above": 1024
        }
      }
    },

File: agent/src/main/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/resourcemanager/RMContextImplEventRunnable.java (hunk 1 above)
Comment (mchataigner) [reply agent/src/main/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/resourcemanager/RMContextImplEventRunnable.java]: Done

File: readers/elasticsearch/src/main/resources/template.json (hunk 2 above)
Comment (mchataigner) [reply readers/elasticsearch/src/main/resources/template.json]: Done

**Pull request merged**